google/cloud/forseti/services/inventory/storage.py

```diff
@@ -764,6 +775,17 @@ def iter(self,
                         Inventory.parent_key == p_key,
                         Inventory.parent_type == p_type,
                         parent_inventory.index == self.index.id)))
+        elif with_resource:
```

Comment:
This is not necessary if dataset policy follow the policy processing flows like iam policy flow.